### PR TITLE
Remove shaperef counter

### PR DESCRIPTION
**Pull request by orange-vertex**
Thus allows carrying the refid through the various conversions (and allows us to break internal code)

**Some of the whitespace was converted to spaces by Qt creator, append ?w=1 to the url of this diff to see without the changes**

### CODE REVIEW

File: salalib/importutils.cpp [hunk 1]
```diff
@@ -262,13 +307,33 @@ namespace depthmapX {
...
     }
+
+    std::map<int, Line> extractLinesWithRef(ColumnData &x1col, ColumnData &y1col, ColumnData &x2col, ColumnData &y2col, ColumnData &refcol) {
+        std::map<int, Line> lines;
+        for(size_t i = 0; i < x1col.size(); i++) {
+            double x1 = stod(x1col[i]);
+            double y1 = stod(y1col[i]);
+            double x2 = stod(x2col[i]);
+            double y2 = stod(y2col[i]);
+            double ref = stoi(refcol[i]);
```
Comment (blsemo) [inline salalib/importutils.cpp]: Does ref need to be a double? Wouldn't an integer be more appropriate?

File: salalib/mgraph.cpp [hunk 2]
```diff
@@ -419,40 +419,40 @@ bool MetaGraph::makeShape(const Line& line)
    return (map.makeLineShape(line,true) != -1);
 }
 
-bool MetaGraph::polyBegin(const Line& line)
+int MetaGraph::polyBegin(const Line& line)
 {
    if (!isEditableMap()) {
       return false;
```
Comment (blsemo) [inline salalib/mgraph.cpp]: That return value needs to be changed - returning `false` for an integer works but is bad form. I'd suggest `-1`.

File: salalib/shapemap.cpp [hunk 3]
```diff
@@ -2873,7 +2937,7 @@ bool ShapeMap::importData(const depthmapX::Table &data, std::vector<int> indices
 
                     if(colcodes.size() >= 32) {
                         for (size_t j = 0; j < column.second.size(); j++) {
-                            m_attributes.setValue(indices[j], colIndex, -1.0f);
+                            m_attributes.setValue(m_attributes.getRowid(indices[j]), colIndex, -1.0f);
```
Comment (blsemo) [inline salalib/shapemap.cpp]: Not sure I understand this change? Isn't set value index based rather than key based?

Comment (orange-vertex) [reply salalib/shapemap.cpp]: Yes, but we are not carrying indices anymore but shape_refs (that's what the various make* functions above return), so now we need to find the index of that ref in the attributes table. I'll just rename the vectors to indicate that they carry shaperefs

**Pull request merged**